### ISSUE TITLE
Cache File Does Not Populate

### ISSUE THREAD

**Issue by ianelletson** [Aedif/TokenVariants#98]
Current version of module, Pathfinder 2e system, and Foundry.

Setting the Cache File under misc (one example shown below, but I've tried multiple) and then clicking "Cache Images" does not create a .json file, nor are any errors logged.

The folder exists; I have tried with and without a pre-existing .json file.

![image](https://user-images.githubusercontent.com/1207114/215273551-69d3287e-7e9f-4ac0-9c7b-2aa01f750a37.png)

![image](https://user-images.githubusercontent.com/1207114/215273567-2ce614ad-a1de-4094-a111-401b2dde1b85.png)

**Comment by Aedif** [comment]
The root of file paths is the `FoundryVTT\Data` folder, so the path should be:
`modules/token-variants/token-variants-cache.json`

**Comment by ianelletson** [comment]
> The root of file paths is the `FoundryVTT\Data` folder, so the path should be: `modules/token-variants/token-variants-cache.json`

I chose a poor example. I meant to have a "token-variants" folder in the `FoundryVTT/Data/token-variants` and put the cache there (so that it isn't removed on module update), see:
![image](https://user-images.githubusercontent.com/1207114/215288828-1e4d45db-59f5-4bbf-83ca-b2ea8bc08ee6.png)

**Comment by Aedif** [comment]
Hmm, strange, I replicated the setup and it works fine for me. Mind exporting the module settings and linking them here?

**Comment by Grandsome** [comment]
My caching also stopped working, I'm using WoD 20th 2.4 and the module version is 4.27. Maybe there's a new incompatibility. 
Here's my module list and the token variant settings:
[PackageList-latest.zip](https://github.com/Aedif/TokenVariants/files/10718015/PackageList-latest.zip)
[token-variants-settings.zip](https://github.com/Aedif/TokenVariants/files/10718020/token-variants-settings.zip)

**Comment by Aedif** [comment]
That's an awful lot of modules 😅. I tested on a fresh world using latest WoD and your TVA settings. The static cache was creating as it should be.

Could you try manually deleting the cache file in the module folder and then running the following in the console:
`game.modules.get("token-variants").api.TVA_CONFIG.debug = true`
Then manually cache the images:
![image](https://user-images.githubusercontent.com/7693704/218505099-d1287793-3bd7-4115-bf2e-2ad73043d04c.png)

The console should now display output similar to this:
![image](https://user-images.githubusercontent.com/7693704/218505254-924d3b6b-b509-402b-b47d-828d191f55d3.png)

Would you mind copy-pasting the output here? Might give me some insight into what's going on here...

**Comment by Grandsome** [comment]
"That's an awful lot of modules 😅" Yeah, I've got a problem.
Here's what the console says after two one hour runs. I'm gonna run find the culprit sometime this week to see if I can find, uh, the culprit.
![image](https://user-images.githubusercontent.com/32459440/218577571-be015c5b-ce84-4f78-b6e7-d0666817c665.png)

![image](https://user-images.githubusercontent.com/32459440/218597682-14bc8473-a373-4074-8029-c98f73c1f98e.png)

**Comment by Aedif** [comment]
Alright the fact that the caching doesn't finish means that the module is either getting stuck in an infinite loop or there is a request that for some reason doesn't time out and the module gets frozen.

My immediate thoughts go towards the s3 buckets. I don't have one setup so I can't test it myself right now but there might have been a change to how s3 file browsing works in Foundry...

If you run this:
```
let path = {
      "source": "s3:vzmtfoundryvtt-store1",
      "text": "newicons/",
      "types": [
        "Item",
        "JournalEntry",
        "Macro",
        "RollTable",
        "Tile"
      ],
      "cache": true
    };
let files = await FilePicker.browse('s3', path.text, {
    bucket: path.source.replace('s3:', ''),
});
console.log(files);
```
What does it return?

**Comment by Grandsome** [comment]
I had my suspicions about the S3 bucket was that was my next test to see if it worked without them, anyway I ran the command it returns:
![image](https://user-images.githubusercontent.com/32459440/218609627-1bd0b1ac-9912-4759-ac18-508f3d443e08.png)

**Comment by Aedif** [comment]
Could you click the arrow to the left of the 'target' to expand the object, I'd like to see if "newicons/" is among the dirs returned. That would be one of the possible ways the modules gets stuck in a loop. Sorry about this, I'd do all of this myself if I had an S3 bucket to test with.

But yes removing the buckets from the paths would be a nice way to quickly rule out that as a problem.

You can always re-import the module settings to restore the paths

**Comment by Grandsome** [comment]
I was able to cache with the latest update, well, maybe it's the update or what I did or both.

I started by unchecking cache for the S3 bucket paths and caching locally first, then I checked cache of my smallest S3 directory to the largest. It seems that icon directory is the problem, it's small in byte size, but the item count is maybe too large.

Or there's a weird permission problem with S3 or some limit on access calls.

I'll just rehost the svgs locally, I put them in remote because in my last game the svgs had a hard time loading for my players even on chrome.

I wonder if reducing the path length would help. 

Anyway, I did a head count: 6 355 objects in the icon directory "newicons", maybe that's too much, current successful cache has 9 613 images (including the same icon directory but local).
